cdap-data-fabric/src/main/java/co/cask/cdap/data/stream/service/InMemoryStreamMetaStore.java

```diff
@@ -51,10 +55,18 @@ public boolean streamExists(String accountId, String streamName) throws Exceptio
   }
 
   @Override
-  public synchronized List<StreamSpecification> listStreams() throws Exception {
-    ImmutableList.Builder<StreamSpecification> builder = ImmutableList.builder();
-    for (String stream : streams.values()) {
-      builder.add(new StreamSpecification.Builder().setName(stream).create());
+  public synchronized Multimap<NamespaceMeta, StreamSpecification> listStreams() throws Exception {
```

Comment:
That was my logic too, we may depend on other fields later. The ``NamespaceMeta`` could evolve to have more information I suppose.